src/engine/libengine/FixedLayoutParser.cpp

```diff
@@ -24,51 +24,10 @@ void FixedLayoutParser::setLayout(QJsonObject l) {
   layout = l;
 }
 
-QString FixedLayoutParser::getCharForKey(int keyNum, bool shift, bool altgr, bool shiftaltgr) {
+QString FixedLayoutParser::processAlphabetKey(int keyNum, bool shift, bool altgr, bool shiftaltgr) {
```

Comment:
Why not just rename this to `getCharForKey` and be with it? :thinking:

Reply:
Yes, I am also not very satisfied with this implementation. I had to do this because we now only check shift modifier when the key is an alphabet key. But when the key is a metacharacter or numeric we need to handle them differently, if you look them closely.

Thanks!